src/js/mqtt.js

```diff
@@ -274,6 +276,18 @@ MqttClient.prototype.disconnect = function(err) {
   this._socket.end();
 };
 
+function getQoS(qos) {
+  return Number.isNaN(qos) ? 0 : (qos >= 0 && qos <= 2 ? qos : 0);
+}
+
+MqttClient.prototype._nextMsgId = function() {
+  if (this._msgId > MAX_MSG_ID) {
+    this._msgId = 1;
```

Comment:
The spec shows the packet id won't be used, that implies us to store the range of identifiers used?

Reply:
Now we have not check the ack id, so it is not very emergency now. Next we need perfect it.

Reply:
Ok, seems good to me now.